In [2]:
# conda install requests
import requests

# GET request
response = requests.get("http://api.open-notify.org/this-api-doesnt-exist") 
print(response.status_code) #endpoint doesnt exist

404


## Status Codes

- `200`: Everything went okay, and the result has been returned (if any).
- `301`: The server is redirecting you to a different endpoint. This can happen when a company switches domain names, or an endpoint name is changed.
- `400`: The server thinks you made a bad request. This can happen when you don’t send along the right data, among other things.
- `401`: The server thinks you’re not authenticated. Many APIs require login credentials, so this happens when you don’t send the right credentials to access an API.
- `403`: The resource you’re trying to access is forbidden: you don’t have the right permissions to see it.
- `404`: The resource you tried to access wasn’t found on the server.
- `503`: The server is not ready to handle the request. 

In [3]:
# using Open Notify API (http://open-notify.org/)

# return data about astronauts in space
response = requests.get("http://api.open-notify.org/astros.json")
print(response.status_code)

200


In [4]:
import json

def jprint(obj):
    # create a formatted string of the Python JSON object
    text = json.dumps(obj, sort_keys=True, indent=4)
    print(text)

jprint(response.json())

{
    "message": "success",
    "number": 7,
    "people": [
        {
            "craft": "ISS",
            "name": "Sergey Ryzhikov"
        },
        {
            "craft": "ISS",
            "name": "Kate Rubins"
        },
        {
            "craft": "ISS",
            "name": "Sergey Kud-Sverchkov"
        },
        {
            "craft": "ISS",
            "name": "Mike Hopkins"
        },
        {
            "craft": "ISS",
            "name": "Victor Glover"
        },
        {
            "craft": "ISS",
            "name": "Shannon Walker"
        },
        {
            "craft": "ISS",
            "name": "Soichi Noguchi"
        }
    ]
}


In [5]:
parameters = {
    "lat": 40.71,
    "lon": -74
}

response = requests.get("http://api.open-notify.org/iss-pass.json", params=parameters)  
jprint(response.json())

{
    "message": "success",
    "request": {
        "altitude": 100,
        "datetime": 1607499033,
        "latitude": 40.71,
        "longitude": -74.0,
        "passes": 5
    },
    "response": [
        {
            "duration": 594,
            "risetime": 1607524821
        },
        {
            "duration": 648,
            "risetime": 1607530588
        },
        {
            "duration": 581,
            "risetime": 1607536467
        },
        {
            "duration": 569,
            "risetime": 1607542344
        },
        {
            "duration": 638,
            "risetime": 1607548163
        }
    ]
}


In [8]:
pass_times = response.json()['response']
jprint(pass_times)

[
    {
        "duration": 594,
        "risetime": 1607524821
    },
    {
        "duration": 648,
        "risetime": 1607530588
    },
    {
        "duration": 581,
        "risetime": 1607536467
    },
    {
        "duration": 569,
        "risetime": 1607542344
    },
    {
        "duration": 638,
        "risetime": 1607548163
    }
]


In [11]:
risetimes = []

for d in pass_times:
    time = d['risetime']
    risetimes.append(time)

print(risetimes)

from datetime import datetime

times = []

for rt in risetimes:
    time = datetime.fromtimestamp(rt)
    times.append(time)
    print(time)

[1607524821, 1607530588, 1607536467, 1607542344, 1607548163]
2020-12-09 22:40:21
2020-12-10 00:16:28
2020-12-10 01:54:27
2020-12-10 03:32:24
2020-12-10 05:09:23
